db/migrate/20210623231309_add_unique_index.rb

```diff
@@ -0,0 +1,5 @@
+class AddUniqueIndex < ActiveRecord::Migration[6.1]
+  def change
+    add_index :absolute_identifiers, [:prefix, :suffix, :pool_identifier], unique: true, name: "absolute_identifiers_uniqueness"
```

Comment:
Clever -- enforce uniqueness on the value by defining an index, since there's not really a column you can put a constraint on.